### PR TITLE
fix: controller.sidecars.additionalSidecarContainers renaming and add tests

### PR DESCRIPTION
**Pull request by jonesbusy**
Fix controller.sidecars.additionalSidecarContainers 

subsume https://github.com/jenkinsci/helm-charts/pull/1001

Missing unit test didn't catch this regression so adding one

### What does this PR do?

Fix controller.sidecars.additionalSidecarContainers renaming and add tests

If you modified files in the `./charts/jenkins/` directory, please also include the following:

```[tasklist]
### Submitter checklist
- [x] I bumped the "version" key in `./charts/jenkins/Chart.yaml`.
- [x] I added a new changelog entry to `./charts/jenkins/CHANGELOG.md`.
- [x] I followed the [technical requirements](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#technical-requirements).
```

### CODE REVIEW

**Review by kvanzuijlen — APPROVED** [review]
LGTM @timja

**Comment by kvanzuijlen** [comment]
@jonesbusy can you please add
`/github/workspace/charts/jenkins/unittests/jenkins-controller-statefulset-test.yaml:hashicorp-tf-password:666`
And `/github/workspace/charts/jenkins/unittests/jenkins-controller-statefulset-test.yaml:hashicorp-tf-password:665` to the `.gitleaksignore` file? I'll put in a PR later today to disable gitleaks altogether.

**Comment by jonesbusy** [comment]
> @jonesbusy can you please add `/github/workspace/charts/jenkins/unittests/jenkins-controller-statefulset-test.yaml:hashicorp-tf-password:666` And `/github/workspace/charts/jenkins/unittests/jenkins-controller-statefulset-test.yaml:hashicorp-tf-password:665` to the `.gitleaksignore` file? I'll put in a separate PR later today to disable gitleaks altogether.

Done

File: charts/jenkins/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ Use the following links to reference issues, PRs, and commits prior to v2.6.0.
 The changelog until v1.5.7 was auto-generated based on git commits.
 Those entries include a reference to the git commit to be able to get more details.
 
+## 5.0.12
+
+* Fix controller.sidecars.additionalSidecarContainers renaming and add tests
```
Comment (timja) [inline charts/jenkins/CHANGELOG.md:17]:
```suggestion
Fix controller.sidecars.additionalSidecarContainers renaming and add tests
```

**Review by timja — APPROVED** [review]

**Pull request merged**